src/game_controller.py

```diff
@@ -86,20 +87,25 @@ def run_bot(self, pick_corpse: bool = False):
         else:
             if self._config.general["info_screenshots"]:
                 cv2.imwrite("./info_screenshots/info_could_not_recover_" + time.strftime("%Y%m%d_%H%M%S") + ".png", self.screen.grab())
-            Logger.error(
-                f"{self._config.general['name']} could not recover from a max game length violation. Restarting the Game.")
-            if self._config.general["custom_message_hook"]:
-                messenger.send_message(f"{self._config.general['name']}: got stuck and will now restart D2R")
-            if restart_game(self._config.general["d2r_path"]):
-                self.game_stats.log_end_game(failed=max_game_length_reached)
-                if self.setup_screen():
-                    self.start_health_manager_thread()
-                    self.start_death_manager_thread()
-                    self.game_recovery = GameRecovery(self.screen, self.death_manager, self.template_finder)
-                    return self.run_bot(True)
-            Logger.error(f"{self._config.general['name']} could not restart the game. Quitting.")
-            messenger.send_message("Got stuck and could not restart the game. Quitting.")
-                
+            if self._config.general['restart_d2r_when_stuck']:
+                Logger.error(
+                    f"{self._config.general['name']} could not recover from a max game length violation. Restarting the Game.")
+                if self._config.general["custom_message_hook"]:
+                    messenger.send_message(f"{self._config.general['name']}: got stuck and will now restart D2R")
+                if restart_game(self._config.general["d2r_path"]):
+                    self.game_stats.log_end_game(failed=max_game_length_reached)
+                    if self.setup_screen():
+                        self.start_health_manager_thread()
+                        self.start_death_manager_thread()
+                        self.game_recovery = GameRecovery(self.screen, self.death_manager)
+                        return self.run_bot(True)
+                Logger.error(f"{self._config.general['name']} could not restart the game. Quitting.")
+                messenger.send_message("Got stuck and could not restart the game. Quitting.")
+            else:
+                Logger.error(
+                    f"{self._config.general['name']} could not recover from a max game length violation. Quitting botty.")
+                if self._config.general["custom_message_hook"]:
+                    messenger.send_message(f"{self._config.general['name']}: got stuck and will now quit botty")
```

Comment:
No need to add `{self._config.general['name']}`. The api implementations take care when to add the name and when not.
I know, in some places we still need to remove it. It currently will send the name twice.